src/infra/marketplace/marketplaceQueryRepo.go

```diff
@@ -31,44 +31,44 @@ func NewMarketplaceQueryRepo(
 	}
 }
 
-func (repo *MarketplaceQueryRepo) parseCatalogItemServices(
+func (repo *MarketplaceQueryRepo) catalogItemServicesFactory(
 	catalogItemServices interface{},
 ) (serviceNamesWithVersions []valueObject.ServiceNameWithVersion, err error) {
-	rawServices, assertOk := catalogItemServices.([]interface{})
+	rawServiceNamesWithVersion, assertOk := catalogItemServices.([]interface{})
 	if !assertOk {
-		return serviceNamesWithVersions, errors.New("InvalidCatalogItemServices")
+		return serviceNamesWithVersions, errors.New(
+			"InvalidCatalogItemServicesStructure",
+		)
 	}
 
-	for _, rawService := range rawServices {
-		rawServiceNameWithVersion, assertOk := rawService.(string)
-		if !assertOk {
-			slog.Error("InvalidCatalogItemService", slog.Any("service", rawService))
-			continue
-		}
-
+	for _, rawServiceNameWithVersion := range rawServiceNamesWithVersion {
 		serviceNameWithVersion, err := valueObject.NewServiceNameWithVersionFromString(
 			rawServiceNameWithVersion,
 		)
 		if err != nil {
 			slog.Error(
 				err.Error(),
-				slog.String("serviceNameWithVersion", rawServiceNameWithVersion),
+				slog.Any("serviceNameWithVersion", rawServiceNameWithVersion),
 			)
 			continue
```

Comment:
If it's a for that will skip the iteration if it fails, than this is a Debug log entry, not an error. Error entries is for errors. If you're skipping, it's not important, then, it's a Debug.